pkg/rancher/migrations.go

```diff
@@ -504,6 +512,49 @@ func migrateMachinePoolsDynamicSchemaLabel(w *wrangler.Context) error {
 	return createOrUpdateConfigMap(w.Core.ConfigMap(), cm)
 }
 
+func migrateSystemAgentDataDirectory(w *wrangler.Context) error {
+	cm, err := getConfigMap(w.Core.ConfigMap(), migrateSystemAgentVarDirToDataDirectory)
+	if err != nil || cm == nil {
+		return err
+	}
+
+	if cm.Data[systemAgentVarDirMigratedKey] == "true" {
+		return nil
+	}
+
+	provClusters, err := w.Provisioning.Cluster().List("", metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, cluster := range provClusters.Items {
+		systemAgentDataDir := ""
+		envVars := make([]rkev1.EnvVar, 0, len(cluster.Spec.AgentEnvVars))
+		for _, e := range cluster.Spec.AgentEnvVars {
+			if e.Name == "CATTLE_AGENT_VAR_DIR" {
```

Comment:
nit: use the global variable in pkg/capr/installer/installer.go,  instead of hard-coded value.